plugins/module_utils/attachment.py

```diff
@@ -0,0 +1,175 @@
+
+        offset = 0
+        total = 1  # Dummy value that ensures loop executes at least once
+        result = []
+
+        while offset < total:
+            response = self.client.get(
+                _path(), query=dict(base_query, sysparm_offset=offset)
+            )
+
+            result.extend(response.json["result"])
+            total = int(response.headers["X-Total-Count"])
+            offset += self.batch_size
+
+        return result
+
+    def get_record(self, query, must_exist=False):
+        records = self.list_records(query)
+
+        if len(records) > 1:
+            raise errors.ServiceNowError(
+                "{0} attachments match the {1} query.".format(len(records), query)
+            )
+
+        if must_exist and not records:
+            raise errors.ServiceNowError(
+                "No attachments match the {0} query.".format(query)
+            )
+
+        return records[0] if records else None
+
+    def create_record(self, metadata, data, check_mode, mime_type):
+        if check_mode:
+            # Approximate the result using the payload and data.
+            return metadata
+        return self.client.request_binary(
+            "POST", _path("file"), mime_type, bin_data=data, query=(metadata or {})
+        ).json["result"]
+
+    def upload_record(self, payload, file_dict, check_mode=False):
+        # "payload" is a dict with defined "table_name", "table_sys_id". This is defined by the module directly.
+        #
+        # "file_dict" is a dict with a mandatory key "path" and optional "name", "type" and "encryption_context".
+        # These properties can be read directly from the yaml attachment list and can be set by the user.
+        new_payload = payload.copy()
+        new_payload["file_name"] = get_file_name(file_dict)
+        file_type = get_file_type(file_dict)
+
+        if (
+            "encryption_context" in file_dict
+            and file_dict["encryption_context"] is not None
+        ):
+            new_payload["encryption_context"] = file_dict["encryption_context"]
+
+        data = None
+        if not check_mode:
+            with open(file_dict["path"], "rb") as file_obj:
+                data = file_obj.read()
+        return self.create_record(new_payload, data, check_mode, file_type)
+
+    def upload_records(self, payload, file_dict_list, check_mode):
+        return [
+            self.upload_record(payload, file_dict, check_mode)
+            for file_dict in (file_dict_list or [])
+        ]
+
+    def delete_record(self, record, check_mode, silent=False):
+        if not check_mode:
+            if record is None and silent:
+                return {"changed": False, "msg": "Skipped. Record doesn't exist."}
+            else:
+                try:
+                    self.client.delete(_path(record["sys_id"]))
+                    return {"changed": True}
+                except errors.UnexpectedAPIResponse:
+                    return {"changed": False}
```

Comment:
We should not silently ignore that because this probably means that something went wrong on the backend (and in this case we do want to see the error) or someone is operating on the same records at the same time as we do. I would suggest that we just let the error bubble up the stack in this case.